chain-signatures/node/src/http_client.rs

```diff
@@ -202,9 +228,23 @@ impl MessageQueue {
                     errors.push(err);
                 } else {
                     compacted += msgs.len();
+                    let send_latency = start.elapsed().as_millis() as f64;
                     crate::metrics::SEND_ENCRYPTED_LATENCY
                         .with_label_values(&[account_id.as_str()])
-                        .observe(start.elapsed().as_millis() as f64);
+                        .observe(send_latency);
+
+                    for (id, msg_timestamp) in presignature_msg_creation_time {
```

Comment:
nit: you shadow the `id` variable of the parent loop, which makes it a bit hard to read. Consider renaming to something like participant_id or node_id and protocol_id or msg_id etc.